src/main/java/redgatesqlci/Utils.java

```diff
@@ -5,34 +5,71 @@
 import hudson.Proc;
 import hudson.model.AbstractBuild;
 import hudson.model.BuildListener;
+import hudson.remoting.Callable;
+import hudson.remoting.VirtualChannel;
 
 import java.io.*;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
 
+import org.jenkinsci.remoting.RoleChecker;
+
 public class Utils {
+    private static String GetEnvironmentVariable(final String variableName, VirtualChannel channel)
+    {
+        try {
+            return channel.call(new Callable<String,RuntimeException>(){
```

Comment:
I think we can use [`MasterToSlaveCallable`](http://javadoc.jenkins-ci.org/jenkins/security/MasterToSlaveCallable.html) instead of `Callable` here, following information at [this link](https://wiki.jenkins-ci.org/display/JENKINS/Slave+To+Master+Access+Control#SlaveToMasterAccessControl-I%27maplugindeveloper.WhatshouldIdo%3F).